esmvaltool/diag_scripts/cvdp/cvdp_wrapper.py

```diff
@@ -0,0 +1,140 @@
+"""wrapper diagnostic for the NCAR CVDP (p)ackage."""
+import logging
+import os
+import re
+import shutil
+import subprocess
+
+from esmvaltool._task import DiagnosticError
+from esmvaltool.diag_scripts.shared import (group_metadata, run_diagnostic,
+                                            select_metadata)
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def setup_driver(cfg):
+    """Write the driver.ncl file of the cvdp package."""
+    cvdp_root = os.path.join(os.path.dirname(__file__), 'cvdp')
+    if not os.path.isdir(cvdp_root):
+        raise DiagnosticError("CVDP is not available.")
+
+    settings = {
+        'outdir': "{0}/".format(cfg['work_dir']),
+        'obs': 'False',
+        'zp': os.path.join(cvdp_root, "ncl_scripts/"),
+        'run_style': 'serial',
+        'webpage_title': 'CVDP run via ESMValTool'
+    }
+    settings['output_data'] = "True" if _nco_available() else "False"
+
+    def _update_settings(line):
+
+        for key, value in settings.items():
+            pattern = r'\s*{0}\s*=.*\n'.format(key)
+            search_results = re.findall(pattern, line)
+            if search_results == []:
+                continue
+            return re.sub(
+                r'".+?"', '"{0}"'.format(value), search_results[0], count=1)
+
+        return line
+
+    content = []
+    driver = os.path.join(cvdp_root, "driver.ncl")
+
+    with open(driver, 'r') as driver_file:
+        for line in driver_file:
+            content.append(_update_settings(line))
+
+    new_driver = os.path.join(cfg['work_dir'], "driver.ncl")
+
+    with open(new_driver, 'w') as new_driver_file:
+        new_driver_file.write("".join(content))
+
+
+def create_link(cfg, inpath):
+    """Create link for the input file.
+
+    The link matches the naming convention of the cvdp package.
+    Returns the path to the link.
+
+    cfg: configuration dict
+    inpath: path to infile
+    """
+    def _create_link_name(inpath):
+        tail = os.path.split(inpath)[1]
+        search_result = re.search(r'[0-9]{4}-[0-9]{4}', tail).group(0)
+        return tail.replace(search_result,
+                            "{0}01-{1}12".format(*search_result.split('-')))
+
+    if not os.path.isdir(inpath):
```

Comment:
This gives me the following error:
```bash
Traceback (most recent call last):
  File "/home/bandela/src/esmvaltool_master/esmvaltool/diag_scripts/cvdp/cvdp_wrapper.py", line 140, in <module>
    main(config)
  File "/home/bandela/src/esmvaltool_master/esmvaltool/diag_scripts/cvdp/cvdp_wrapper.py", line 133, in main
    setup_namelist(cfg)
  File "/home/bandela/src/esmvaltool_master/esmvaltool/diag_scripts/cvdp/cvdp_wrapper.py", line 92, in setup_namelist
    links = [create_link(cfg, item["filename"]) for item in value]
  File "/home/bandela/src/esmvaltool_master/esmvaltool/diag_scripts/cvdp/cvdp_wrapper.py", line 92, in <listcomp>
    links = [create_link(cfg, item["filename"]) for item in value]
  File "/home/bandela/src/esmvaltool_master/esmvaltool/diag_scripts/cvdp/cvdp_wrapper.py", line 71, in create_link
    raise DiagnosticError("Path {0} does not exist".format(inpath))
esmvaltool._task.DiagnosticError: Path /home/bandela/esmvaltool_output/recipe_cvdp_20181206_102544/preproc/diagnostic1_default_tas/CMIP5_MPI-ESM-LR_Amon_historical_r1i1p1_T2Ms_tas_2000-2002.nc does not exist
```
I think it should be replaced by
```python
 if not os.path.isfile(inpath):
```

Reply:
Yes, I fixed it